src/shogun/io/streaming/ParseBuffer.h

```diff
@@ -271,8 +278,8 @@ template <class T> CParseBuffer<T>::CParseBuffer(int32_t size)
 	ex_ring = SG_CALLOC(Example<T>, ring_size);
 	ex_used = SG_MALLOC(E_IS_EXAMPLE_USED, ring_size);
 #ifdef HAVE_CXX11
-	read_mutex = std::make_shared<std::mutex>();
-	write_mutex = std::make_shared<std::mutex>();
+//	read_mutex = std::make_shared<std::mutex>();
```

Comment:
I thought reading all of it in one would be better than comparing using git diff for this commit. Will refactor things asap.